compiler/rustc_ast/src/visit.rs

```diff
@@ -720,6 +726,10 @@ pub fn walk_expr<'a, V: Visitor<'a>>(visitor: &mut V, expression: &'a Expr) {
             visitor.visit_anon_const(count)
         }
         ExprKind::Struct(ref path, ref fields, ref optional_base) => {
+            let (maybe_qself, path) = &**path;
```

Comment:
if you want to use patterns for these, you can use `box (ref maybe_qself, ref path)`

Reply:
ooooohhh....

Reply:
Does `box` work with custom smart pointers? `path` is a `&P<..>`.

Reply:
oh... I forgot about that. Sorry. You're right, this doesn't work right now